mopidy_listenbrainz/frontend.py

```diff
@@ -15,14 +19,135 @@ class ListenbrainzFrontend(pykka.ThreadingActor, CoreListener):
+            [
+                (ref.uri, ref)
+                for ref in existing_playlists
+                if ref.uri.startswith(recommendation_playlist_uri_prefix)
+            ]
+        )
+
+        for playlist_data in playlist_datas:
+            source = playlist_data.playlist_id
+            playlist_uri = f"{recommendation_playlist_uri_prefix}:{playlist_data.playlist_id}"
+            tracks = self._collect_playlist_tracks(playlist_data)
+
+            if len(tracks) == 0:
+                logger.debug(
+                    f"Skipping import of playlist with no known track for {source!r}"
+                )
+                continue
+
+            if playlist_uri in filtered_existing_playlists:
+                filtered_existing_playlists.pop(playlist_uri)
+                # must pop since filtered_existing_playlists will
+                # finally be deleted
+
+                logger.debug(f"Already known playlist {playlist_uri}")
+                # maybe there're new tracks in Mopidy's database...
+            else:
+                query = self.playlists.create(
+                    name=playlist_uri, uri_scheme="listenbrainz"
+                )
+                # Hack, hack: The backend uses first parameter as URI,
+                # not name...
+                playlist = query.get()
+                if playlist is None:
+                    logger.warning(f"Failed to create playlist for {source!r}")
+                    continue
+
+                logger.debug(f"Playlist {playlist.uri!r} created")
+
+            complete_playlist = Playlist(
+                uri=playlist_uri,
+                name=playlist_data.name,
+                tracks=tracks,
+                last_modified=playlist_data.last_modified,
+            )
+            query = self.playlists.save(complete_playlist)
+            playlist = query.get()
+            if playlist is None:
+                logger.warning(f"Failed to save playlist for {source!r}")
+            else:
+                import_count += 1
+                logger.debug(
+                    f"Playlist saved with {len(playlist.tracks)} tracks {playlist.uri!r}"
+                )
+
+        for playlist in filtered_existing_playlists.values():
+            logger.debug(f"Deletion of obsolete playlist {playlist.uri!r}")
+            self.playlists.delete(playlist.uri)
+
+        logger.info(
+            f"Successfull import of ListenBrainz playlists: {import_count}"
+        )
+        self._schedule_playlists_import()
+
+    def _collect_playlist_tracks(
+        self, playlist_data: PlaylistData
+    ) -> List[Track]:
+        tracks: List[Track] = []
+        for track_mbid in playlist_data.track_mbids:
+            query = self.library.search(
+                {"musicbrainz_trackid": [track_mbid]}, uris=["local:"]
+            )
+            # search only in local database since other backends can
+            # be quite long to answer, should we offer choice through
+            # config?
```

Comment:
I think this would be useful. For me no playlists are showing up because I use the `mopidy-subidy` extension exclusively

Reply:
@suaviloquence The track search is not limited to Mopidy-Local anymore. The default is to search through all backends. The `search_schemes` configuration key allows to specify a list of schemes to limit the results to URIs with the given schemes.

Reply:
Arggg... I just tried with my living room instance of Mopidy, a "production instance" with multiple backends. And 
some backends seems to not support search by `musicbrainz_trackid` and not respect the API...

No idea how to fix this...
```
mai 14 22:18:07 argos mopidy[1925]: INFO     [ListenbrainzFrontend-15 (_actor_loop)] mopidy_listenbrainz.frontend Importing ListenBrainz playlists
mai 14 22:18:08 argos mopidy[1925]: INFO     [InternetArchiveBackend-6 (_actor_loop)] mopidy_internetarchive.library Not searching Mopidy-InternetArchive: Keyword "musicbrainz_trackid" not supported
mai 14 22:18:08 argos mopidy[1925]: ERROR    [Core-11 (_actor_loop)] mopidy.core.library FileBackend backend returned bad data: Expected a SearchResult instance, not []
mai 14 22:18:08 argos mopidy[1925]: ERROR    [Core-11 (_actor_loop)] mopidy.core.library StreamBackend backend returned bad data: Expected a SearchResult instance, not []
```